src/main/java/baritone/PerformanceCritical.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation that should be used on methods which are performance critical (i.e. called millions of times per second
+ * by the pathfinder) and should be modified with care.
+ *
+ * @author Brady
+ */
+@Target({ElementType.METHOD, ElementType.CONSTRUCTOR})
```

Comment:
Isn't a constructor marked as `@PerformanceCritical` already a hint you are doing something wrong?
I don't think this should be applicable to constructors.

Reply:
True